src/mesido/workflows/io/write_output.py

```diff
@@ -606,22 +695,36 @@ def _name_to_asset(name):
 
             # Calculate the estimated energy source [%] for an area
             try:
-                total_energy_produced_locally_wh_area = total_energy_produced_locally_wh[
-                    subarea.name
-                ]
+                if not np.isnan(total_energy_produced_locally_wh[subarea.name]):
```

Comment:
out of curiosity, how can total_energy_produced_locally_wh[subarea.name] not give a KeyError but be NaN.

Reply:
To be honest, I am also not sure. This happened partner use case that Sam investigated. I realized that the NAN could be the end result of the kpi value if a NAN is passed to this calculation or if we divide by 0. So I added both checks